arelle/plugin/validate/EDINET/PluginValidationDataExtension.py

```diff
@@ -103,6 +104,18 @@ def isCorporateForm(self, modelXbrl: ModelXbrl) -> bool:
             return True
         return False
 
+    @lru_cache(1)
+    def isCorporateReport(self, modelXbrl: ModelXbrl) -> bool:
+        if self.jpcrpNamespace in modelXbrl.namespaceDocs.keys():
+            return True
```

Comment:
Is the pattern to apply `@lru_cache` to all methods?  It otherwise doesn't seem useful for this one.